SYNC-364: add REST point to download the parent db

cioan opened the pull request:
@mseaton , I was able to download the parent db using the new REST point below using the following wget command:

db/malawi  > wget --no-check-certificate \
  --method GET \
  --header 'Authorization: Basic Y2lvYW46Q2lvYW4xMjM=' \
   'https://neno-ci.pih-emr.org/openmrs/ws/rest/v1/sync/downloadDb' -O nenoDb.sql
--2025-03-25 08:50:13--  https://neno-ci.pih-emr.org/openmrs/ws/rest/v1/sync/downloadDb
Resolving neno-ci.pih-emr.org (neno-ci.pih-emr.org)... 74.50.48.97
Connecting to neno-ci.pih-emr.org (neno-ci.pih-emr.org)|74.50.48.97|:443... connected.
HTTP request sent, awaiting response... 200 200
Length: 2041352360 (1.9G) [application/octet-stream]
Saving to: ‘nenoDb.sql’

nenoDb.sql                                        100%[============================================================================================================>]   1.90G  6.58MB/s    in 5m 52s  

2025-03-25 08:57:29 (5.53 MB/s) - ‘nenoDb.sql’ saved [2041352360/2041352360]

mseaton commented on omod/src/main/java/org/openmrs/sync/rest/CreateChildServerRestController.java:
```diff
@@ -0,0 +1,33 @@
+import java.io.InputStream;
+
+@RestController
+public class CreateChildServerRestController {
```
Looks good @cioan .  Maybe rename this class to "DownloadDbRestController" or "GenerateDataFileRestController"?  This doesn't create a child server.

mseaton replied:
This is also a very dangerous endpoint.  We need to ensure it is tightly privileged.

cioan replied:
I just added 'Manage Synchronization' as a required privilege. Let me know if I should use a different privilege? 
I also refactor the controller name.

cioan commented:
@mseaton , I have added a new REST point, registerChildServer which allows us to register a child server via the web services. It basically mirrors the existing configServer.form?type=CHILD :

<img width="931" alt="Screenshot 2025-04-01 at 4 41 28 PM" src="https://github.com/user-attachments/assets/abedd4b6-7a04-4ac1-8d21-f6a517123a75" />

The new REST point accepts a json in the following format:

```javascript
{
	"childServer" : {
		"uuid": "14CD683C-BF22-4D28-B154-6A9947DDC6AB",
		"nickname": "child-7",
		"username": "childSeven",
		"password": "Sync1234",
		"notSendTo": [
			"org.openmrs"			
		],
		"notReceiveFrom": [
			"org.openmrs.notification.Alert",
			"org.openmrs.GlobalProperty",
			"org.openmrs.hl7",
			"org.openmrs.scheduler"
		]
	}
}
``` 
I have tested all different scenarios and I got the expected results:


![Screenshot 2025-04-01 at 1 21 51 PM](https://github.com/user-attachments/assets/7d5b3c7f-6491-4b70-9a4b-3ee5379a6c76)

![Screenshot 2025-04-01 at 1 20 05 PM](https://github.com/user-attachments/assets/bf218132-6995-4fbb-8d59-08ebfb1b0be6)

cioan commented:
@mseaton , per our discussion last week, I have moved the REST end points to pihmalawi, but this small refactoring in the sync module is still needed in order to leverage the new registerChildServer() method I added to the SyncService. Basically, it was just moving the saveOrUpdateServerClasses method from the form controller to the SyncUtils. Thanks!

mseaton commented:
Just make sure you update the PR message / commit message to reflect the updated nature of the change.
